### PR TITLE
worker: add replication

### PR DESCRIPTION
**Pull request by juliangruber**
There's lots left to do, but this is setting up basic replication between hyperdrives.

### CODE REVIEW

File: worker/index.js [hunk 1]
```diff
@@ -1,20 +1,52 @@
...
-    this.p2p = new P2PCommons()
-    this.ready = this.p2p.ready()
   }
 
   async start () {
-    await this.ready
+    const drives = new Map()
+
+    const swarm = hyperswarm({ queue: { multiplex: true } })
+    swarm.on('error', console.error)
```
Comment (juliangruber) [inline worker/index.js]: needs better error handling

File: worker/index.js [hunk 2]
```diff
@@ -1,20 +1,52 @@
...
+    swarm.on('connection', (socket, info) => {
+      socket.setMaxListeners(Infinity)
```
Comment (juliangruber) [inline worker/index.js]: it seems like we're getting redundant connections

File: worker/index.js [hunk 3]
```diff
@@ -1,20 +1,52 @@
...
+      for (const topic of info.topics) {
+        const drive = drives.get(toStr(topic))
+        if (drive) {
+          const isInitiator = info.client
+          pipeline(
+            socket,
+            drive.replicate(isInitiator, { timeout: 0 }),
+            socket,
+            err => err && console.error(err)
```
Comment (juliangruber) [inline worker/index.js]: needs better error handling

File: worker/index.js (hunk 2 above)
Comment (dpaez) [reply worker/index.js]: Maybe here we can use the [deduplication](https://github.com/hyperswarm/hyperswarm#dropped--infodeduplicatelocalidbuffer-remoteidbuffer) feature from hyperswarm.

**Comment by dpaez** [comment]
Looks great @juliangruber 

This is useful to start testing things.

**Pull request merged**